src/server_config.c

```diff
@@ -5870,6 +5870,41 @@ nc_server_config_cert_exp_notif_thread_wakeup(void)
 
 #endif /* NC_ENABLED_SSH_TLS */
 
+/**
+ * @brief Wait for any pending updates to complete, then mark the server as "applying configuration".
+ *
+ * @return 0 on success, 1 on timeout.
+ */
+static int
+nc_server_config_update_start(void)
```

Comment:
Okay, I would prefer using a conditional variable but that would require more changes since you would need a mutex instead of a rwlock. So at least remove the infinite-waiting `rwlock` please, there should be no such calls in libraries.

Reply:
Done.